### DISCUSSION TITLE
Date Range "Easy Button"

### DISCUSSION THREAD

**Discussion by jdbranham** [apache/superset#19798] [Ideas]
My client wants an easy button for a date range filter.

The current date range filter takes too many clicks to select a range, and they are confused about the exclusive end date.
Ideally, we can create a dashboard and add this new filter as a native filter. The filter should show calendar date pickers for start and end date [always 00:00 for time], and the end date should be inclusive .

I've talked with several business/executive folks, and this type of filter would satisfy them.

Anyone available to build this within the next 3 weeks?
I've got $$ for you! =)

**Comment by jdbranham** [comment]
This is done! Working with a developer to get a PR up soon.

**Reply by kasiazjc** [reply]
Thanks for working on it! 
Could you share a link to PR when it's done? Curious to see the improvements!

**Reply by sfirke** [reply]
@jdbranham is this functionality still something you'd consider merging back to the main Superset branch?

**Reply by jdbranham** [reply]
Here is the experiment I was doing (incomplete)
https://github.com/apache/superset/compare/master...savantly-net:superset:jb/simple-date-range-picker

I contracted @BharatNischal to build this one. It worked for the specific version, but it needs to be updated and brought in as a new plugin to main, or maybe just modify the existing Calendar plugin to have similar functionality.
Unfortunately, my users had a hard time adapting to a BI tool, so I ended up creating another dashboard application from scratch.
https://github.com/BharatNischal/Calendar-range-plugin/tree/main/src

**Reply by DHIA2010** [reply]
any update about when this feature will be ready to use ?

thank you @jdbranham for working on this it really needed

**Reply by alexmattos** [reply]
Hi @jdbranham  - we have a similar use case where a simpler date picker is needed.  This this move forward or did you stick with your developed solution and this was dropped?

**Reply by jdbranham** [reply]
Unfortunately, the learning curve was too great for the specific client that we developed this for.
We ended up switching to a custom developed reporting/analytics app.
We still have Superset available, for a few BI users, but the calendar range plugin was only compatible with an old version.

Ideally, I think a component that could be configured by a "date range supplier" would satisfy a lot of use-cases.
Otherwise, I've ended up writing lots of Jinja to get more custom functionality - fiscal date ranges are a good example.

**Comment by treygilliland** [comment]
It's kind of crazy this isn't a feature by now. Most users don't want to use antd's date selector and select times. They just want to pick a day on a calendar and don't really understand much beyond that

Our organization forked v2.0.1 to add a simple date selector in and still use that version to this day. I'd love to migrate to v4, but don't have the bandwidth to take it on atm

**Comment by jose-vines-toteat** [comment]
It's amazing to me that superset has an incredibly flexible time filter, but not a simple one. Every one of my clients asks for a simple calendar start/end selector :/
